bridge/client/app/_components/ktb-sequence-list/ktb-sequence-list.component.html

```diff
@@ -19,21 +19,24 @@ <h2>Sequences</h2>
         </div>
       </ng-container>
       <ng-template #trace>
-        <ng-container *ngIf="isTrace(row) as trace">
+        <ng-container *ngIf="isSubsequence(row) as subSequence">
           <dt-icon
-            *ngIf="!trace.isLoading() || trace.isApproval() || trace.getLastTrace().isApproval(); else loading"
+            *ngIf="subSequence.state === 'finished' || subSequence.hasPendingApproval; else loading"
             class="event-icon"
-            [class.error]="trace.isFaulty()"
-            [class.success]="trace.isSuccessful()"
-            [class.highlight]="trace.isApproval() || trace.getLastTrace().isApproval()"
-            [name]="trace.isFinished() ? trace.getIcon() : trace.getLastTrace().getIcon()"
+            [class.error]="subSequence.result === 'fail'"
+            [class.success]="subSequence.result === 'pass'"
```

Comment:
I would introduce methods here, `isFaulty()` and `isSuccessful()` or similar, because it reduces necessary changes if the model changes and also use the `ResultTypes` enum.

Reply:
Because we just have one single property here, I would just use equals. Nevertheless I'll use the enum here